prov/verbs/src/verbs_cm.c

```diff
@@ -194,6 +194,47 @@ fi_ibv_msg_ep_accept(struct fid_ep *ep, const void *param, size_t paramlen)
 	return 0;
 }
 
+static void *fi_ibv_msg_alloc_xrc_params(const void *param, size_t *paramlen)
+{
+	struct fi_ibv_xrc_cm_data *cm_data;
+	size_t cm_datalen = sizeof(*cm_data) + *paramlen;
+
+	if (cm_datalen > FI_IBV_CM_DATA_SIZE) {
+		VERBS_WARN(FI_LOG_EP_CTRL, "XRC CM private data overflow\n");
+		errno = FI_EINVAL;
+		return NULL;
+	}
+
+	cm_data = malloc(cm_datalen);
+	if (!cm_data) {
+		errno = FI_ENOMEM;
+		return NULL;
+	}
+	if (paramlen)
+		memcpy((cm_data + 1), param, *paramlen);
+	*paramlen = cm_datalen;
```

Comment:
Should the assignment of paramlen be done inside of the if check?  Otherwise, you are dereferencing a NULL pointer.

Reply:
I think the actual thing that should be done is the if statement should dereference the paramlen pointer. This is an internal function and paramlen will always be non-Null.  If you are fine with that I'll fold that in.

Reply:
Then what is the use of the if check?  If it will always be non-Null then why check for it?

Reply:
The pointer will be non-NULL, the value it points to could be 0. So avoid stacking parameters for a zero length memcpy was the intent.

Reply:
With that said I'm fine to take the check out as it is not necessary. Why don't I proceed that way.

Reply:
You are checking the paramlen address not its value in the if check.
So is that what you meant by your first comment, "I think the actual thing that should be done is the if statement should dereference the paramlen pointer"? 
So the if statement should be "if (*paramlen)"?
If this is the correction, then I am okay.

Reply:
It is. The existing statement should either be corrected to "if (*paramlen)" or the checked should be removed and just let memcpy not copy. I'm fine with either.

Reply:
I removed the check altogether since it is not required. Thanks.